PyInstaller/building/imphook.py

```diff
@@ -12,18 +12,449 @@
 Code related to processing of import hooks.
 """
 
-import glob
+import glob, weakref
 import os.path
 
 from .. import log as logging
-from .utils import format_binaries_and_datas
-from ..compat import expand_path
-from ..compat import importlib_load_source
+from ..compat import expand_path, importlib_load_source, FileNotFoundError
 from .imphookapi import PostGraphAPI
+from .utils import format_binaries_and_datas
 
 logger = logging.getLogger(__name__)
 
 
+class ModuleHookCache(dict):
```

Comment:
Inheriting from the built-in collection types (`dict`, `list`, and `set`) is discouraged in Python 3. The builtin types are more aggressively optimized, and will return an instance of the base type in cases where you would expect an instance of the subclass. For `dict` this is only an issue when you call `copy()`, which always returns a `dict` instead of the subclass.

It is recommended to inherit from the base classes in the `collections` module instead - in this case `collections.UserDict`